joomla/joomla-cms
# [4.0] Display Custom Fields for Frontend Contact View Conditionally

Pull Request for Issue #33801

### Summary of Changes
Refactored the foreach block. The loop iteration is executed if the field's value exists and its parent Field Group is to be displayed.

### Testing Instructions

1. Activate user profile
2. Create a contact
3. Set display user profile
4. Create a custom field in users (Multiple fields under multiple field groups are recommended for thorough testing)
5. Let the field empty in the user's profile
6. Create a contact menu item
7. Visit the contact created from Frontend

Before the PR, you will be able to see the field groups for all possible groups, including those left blank.
After the PR, Only those groups will be shown that have a field with a value.

<hr>

### Actual result BEFORE applying this Pull Request
![image](https://user-images.githubusercontent.com/53610833/117937620-d4c77200-b323-11eb-9e12-b69705539215.png)



### Expected result AFTER applying this Pull Request
![image](https://user-images.githubusercontent.com/53610833/117937303-73070800-b323-11eb-8391-2b43e2c46732.png)



### Documentation Changes Required
None
